src/libraries/System.Reflection/tests/TypeInfoTests.cs

```diff
@@ -471,10 +471,10 @@ public static void IsEnumDefined(object value, bool expected)
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/mono/mono/issues/15028", TestRuntimes.Mono)]
         public void IsEnumDefined_Invalid()
         {
-            AssertExtensions.Throws<ArgumentException>("", () => typeof(NonGenericClassWithNoInterfaces).GetTypeInfo().IsEnumDefined(10));
+            AssertExtensions.Throws<ArgumentException>("enumType", () => typeof(NonGenericClassWithNoInterfaces).GetTypeInfo().IsEnumDefined(10));
+            AssertExtensions.Throws<ArgumentException>("enumType", () => typeof(NonGenericClassWithNoInterfaces).GetTypeInfo().IsEnumDefined("10"));
```

Comment:
If I am reading the code correctly, this is going to fail (the argument name is not going to be `enumType`).

It may be best to do the check on the managed side in C#, for both cases. And change the `FCThrowArgument` in the unmanaged implementation to just `_ASSERTE(th.IsEnum())`. More code in C# is good, one line at a time :-)

```
            if (!IsEnum)
                throw new ArgumentException(SR.Arg_MustBeEnum, "enumType");
```

Reply:
@jkotas, Looking at [Type.Enum.cs](https://github.com/dotnet/runtime/blob/0a2f0e6da98a784601b37d19417b930a33b2c478/src/libraries/System.Private.CoreLib/src/System/Type.Enum.cs#L26), the check is there. Not sure how it didn't throw in that line and the code flow went to the cpp implementation. I'm not very familiar with the code base, so excuse me if the question is looking trivial.

Reply:
Oh, got it. The check exists in Type.Enum.cs but is missing in RuntimeType.cs, will fix that.

Reply:
@jkotas, A side question, should **all** argument exceptions through the code base have ParamName specified? or there are cases where it shouldn't be specified?

Reply:
There are cases where it does not make sense. For example, if there are multiple arguments and it is not clear which one to blame for having wrong value.